src/main/scala/com/comcast/xfinity/sirius/uberstore/segmented/SegmentedCompactor.scala

```diff
@@ -92,17 +105,19 @@ object SegmentedCompactor {
   }
 
   /**
-   * Compact a single Segment, according to the GC keys provided
+   * Compact a single Segment, according to the GC keys provided.  Deletes that are older than the maxDeleteAge
+   * are purged.
    *
    * @param keys keys to gc against
    * @param source source segment, to be GC'ed
    * @param dest destination segment, empty, will be populated
    */
   private def compactSegment(keys: Set[String], source: Segment, dest: Segment) {
-    // XXX if we want to remove old deletes from the log...
-    // val currentTime = System.currentTimeMillis
+    val currentTime = System.currentTimeMillis
+    val earliestDelete = if (maxDeleteAgeMillis > currentTime) 0L else currentTime - maxDeleteAgeMillis
+
     source.foreach {
-      // case OrderedEvent(_, timestamp, Delete(key)) if (currentTime - timestamp) > MAX_DELETE_AGE => // nothing
+      case OrderedEvent(_, timestamp, Delete(key)) if timestamp < earliestDelete => // nothing
```

Comment:
I do like using `earliestDelete` so we don't have to see that comparison happen down here.